File: 0027-remove-element/0027-remove-element.py

class Solution(object):
    def removeElement(self, nums, val):
        l , r = 0, len(nums) - 1
        
        if not nums:
            return 0
        
        while l < r:
            if nums[r] == val:
                    r-=1
            else:
                if nums[l] == val:
                    tmp = nums[l]
                    nums[l] = nums[r]
                    nums[r] = tmp
                    r-=1
                l+=1
        
        if nums[l] == val:
            return l
        else:
            return l+1

File: 0027-remove-element/0027-remove-element.c
int removeElement(int* nums, int numsSize, int val) {

    int k = 0;

    for (int r = 0; r < numsSize; r++) {
        if (nums[r] != val) {
            nums[k] = nums[r];
            k += 1;
        }
    }
    return k;
    
}